app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-link-click-tracker.js

```diff
@@ -0,0 +1,156 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+  GOVUK.analyticsGA4 = GOVUK.analyticsGA4 || {}
+
+  GOVUK.analyticsGA4.linkClickTracker = {
+
+    trackLinkClicks: function () {
+      this.internalLinksDomain = 'www.gov.uk/'
+      document.querySelector('body').addEventListener('click', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('contextmenu', this.handleClick.bind(this))
+      document.querySelector('body').addEventListener('mousedown', this.handleMousedown.bind(this))
+    },
+
+    handleClick: function (event) {
+      var element = event.target
+
+      if (element.tagName !== 'A') {
+        element = element.closest('a')
+      }
+
+      if (!element) {
+        return
+      }
+
+      var clickData = {}
+      var href = element.getAttribute('href')
+
+      if (this.isMailToLink(href)) {
```

Comment:
Each of the three functions in this `if` (`isMailtoLink`, `isDownloadLink` and `isExternalLink`) first check for the existence of `href` and return false if it's not there. I think it would be more efficient to have that check once, here, prior to the `if` - if the script can't find a `href` attribute then it might as well finish here.